testapps/test_pillowtop/tests/test_xform_pillow.py

```diff
@@ -161,6 +162,46 @@ def test_app_metadata_tracker_synclog_processed(self):
         )
         self.assertEqual(last_submission.app_id, self.metadata.app_id)
 
+    def _test_heartbeat(self, num_submissions):
+        sync_date = datetime.utcnow()
+        UserReportingMetadataStaging.add_heartbeat(
+            self.domain, self.user._id, self.metadata.app_id,
+            '123', sync_date, 'heartbeat_device_id',
+            230, 2, 10, 'CommCare 2.28', 'en'
+        )
+
+        self.assertEqual(UserReportingMetadataStaging.objects.count(), 1)
+        self.assertEqual(UserReportingMetadataStaging.objects.first().user_id, self.user._id)
+
+        process_reporting_metadata_staging()
+        self.assertEqual(UserReportingMetadataStaging.objects.count(), 0)
+        ccuser = CommCareUser.get_by_user_id(self.user._id, self.domain)
+
+        self.assertEqual(len(ccuser.reporting_metadata.last_submissions), num_submissions)
+        self.assertEqual(len(ccuser.reporting_metadata.last_syncs), 1)
+        self.assertEqual(ccuser.reporting_metadata.last_syncs[0].sync_date, sync_date)
+        self.assertEqual(ccuser.reporting_metadata.last_sync_for_user.sync_date, sync_date)
+        self.assertEqual(ccuser.last_device.device_id, 'heartbeat_device_id')
+        app_meta = ccuser.last_device.get_last_used_app_meta()
+        self.assertEqual(app_meta.num_unsent_forms, 2)
+        self.assertEqual(app_meta.num_quarantined_forms, 10)
+
+    @run_with_all_backends
```

Comment:
why do these need to be run on all backends?

Reply:
Perhaps not necessary for testing just heartbeat. Removing